client/src/components/Settings/SettingsContributing.res

```diff
@@ -1,68 +1,259 @@
 // open Tc
-
+open Belt_extended
 module Utils = SettingsUtils
 
-let title = "Contributing"
-@ppx.deriving(show)
-type rec t = {tunnelHost: Utils.formField}
+// -------------------
+// Tunnel URL form
+// -------------------
 
-let default = {tunnelHost: Utils.defaultFormField}
+// This is a single input box that we save/load via API
+module TunnelHost = {
+  // UX:
```

Comment:
```suggestion
  // UX TODOs:
```

Reply:
These are done. I listed the UX as a description of how it should work